.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,48 @@
+# this build is designed to replicate the Travis CI workflow
+name: Build with Node
+
+on:
+  push:
+    branches: [ $default-branch, 'github-actions' ]
+  pull_request:
+    branches: [ $default-branch ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [ 13.x ]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - run: npm ci
+      - run: npm run build --if-present
+      - run: npm test
+
+      - name: Coveralls
+        uses: coverallsapp/github-action@master
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set up JDK 1.8 and Maven settings file
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+          server-id: openmrs-repo-modules
+          server-username: MAVEN_REPO_USERNAME
+          server-password: MAVEN_REPO_PASSWORD
+
+      - name: Deploy release
+        run: mvn deploy:deploy-file -Pdeploy-snapshot
+        env:
+          MAVEN_REPO_USERNAME: pih
+          MAVEN_REPO_PASSWORD: ${{ secrets.MAVEN_REPO_PASSWORD }}
```

Comment:
@ibacher / @cintiadr / @bmamlin - I'll want to have a MAVEN_REPO_PASSWORD and MAVEN_REPO_USERNAME that we can use for deployment, ideally as organization-wide secrets that are shared with all of our OWA projects.  Can we do this?

Reply:
Follow-up, since I have access to the main openmrs repos, I configured this with my personal credentials and an API key for my account temporarily.  This seems to work, and on my last push to this feature branch this successfully deployed to the owa-snapshots repository.  Thanks for setting that up @ibacher .